xarray/backends/zarr.py

```diff
@@ -666,10 +666,23 @@ def open_store(
             use_zarr_fill_value_as_mask=use_zarr_fill_value_as_mask,
             zarr_format=zarr_format,
         )
-        group_paths = list(_iter_zarr_groups(zarr_group, parent=group))
+        from zarr import Group
+
+        group_members: dict[str, Group]
+        if _zarr_v3():
+            group_members = {
+                f"{group}/{path}" if group != "/" else f"/{path}": store
+                for path, store in dict(zarr_group.members(max_depth=None)).items()
+                if isinstance(store, Group)
+            }
+            group_members[group] = zarr_group
+        else:
+            group_paths = list(_iter_zarr_groups(zarr_group, parent=group))
+            group_members = {path: zarr_group.get(path) for path in group_paths}
```

Comment:
```suggestion
            group_members = {path: zarr_group[path] for path in group_paths}
```

This is slightly safer in so far as it will raise a key error if we've returned an invalid path.

Reply:
This was done in my fixes PR